index/scorch/segment/zap/segment.go

```diff
@@ -418,13 +418,29 @@ func (s *SegmentBase) DocNumbers(ids []string) (*roaring.Bitmap, error) {
 		}
 
 		postingsList := emptyPostingsList
-		for _, id := range ids {
+		filteredIds := ids[:0]
```

Comment:
oh, this is nice to reuse the incoming slice backing array.

but, should double check that all the callers of DocNumbers() are ok with their input ids[] getting modified?

Reply:
I got rid of this. Even if current callers are ok, we cannot be sure about future callers.